src/vconsole/vconsole-setup.c

```diff
@@ -305,36 +300,109 @@ static void setup_remaining_vcs(int fd, bool utf8) {
+                /*
+                 * copy unicode translation table
+                 * unimapd is a ushort count and a pointer to an
+                 * array of struct unipair { ushort, ushort }
+                 */
                 r = ioctl(fd_d, PIO_UNIMAPCLR, &adv);
                 if (r < 0) {
-                        log_warning_errno(errno, "PIO_UNIMAPCLR failed, unimaps might be incorrect for tty%i: %m", i);
+                        log_warning_errno(errno, "PIO_UNIMAPCLR failed, unimaps might be incorrect for tty%u: %m", i);
                         continue;
                 }
 
                 r = ioctl(fd_d, PIO_UNIMAP, &unimapd);
                 if (r < 0) {
-                        log_warning_errno(errno, "PIO_UNIMAP failed, unimaps might be incorrect for tty%i: %m", i);
+                        log_warning_errno(errno, "PIO_UNIMAP failed, unimaps might be incorrect for tty%u: %m", i);
                         continue;
                 }
 
                 log_debug("Font and unimap successfully copied to %s", ttyname);
         }
 }
 
+static int find_source_vc(char **ret_path, unsigned *ret_idx) {
+        char *path;
+        unsigned i;
+        int ret_fd, loop_errno;
+
+        path = new(char, sizeof("/dev/tty63"));
+        if (path == NULL)
+                return log_oom();
+
+        for (i = 1; i <= 63; i++) {
+                _cleanup_close_ int fd = -1;
+
+                if (is_allocated(i) < 0) {
+                        loop_errno = errno;
+                        continue;
+                }
+
+                sprintf(path, "/dev/tty%u", i);
+                fd = open_terminal(path, O_RDWR|O_CLOEXEC|O_NOCTTY);
+                if (fd < 0) {
+                        loop_errno = errno;
+                        continue;
+                }
+                if (is_settable(fd) < 0) {
+                        loop_errno = errno;
+                        continue;
+                }
+
+                /* all checks passed, return this one as a source console */
+                *ret_idx = i;
+                *ret_path = path;
+                ret_fd = fd;
+                fd = -1;
+                return ret_fd;
+        }
+
+        errno = loop_errno;
+        free(path);
+        return log_error_errno(errno, "No usable source console found, last error for /dev/tty%u was: %m", i - 1);
+}
+
+static int test_source_vc(char **ret_path, const char *src_vc) {
+        char *path;
+        _cleanup_close_ int fd = -1;
+        int ret_fd;
+
+        fd = open_terminal(src_vc, O_RDWR|O_CLOEXEC|O_NOCTTY);
+        if (fd < 0)
+                return log_error_errno(fd, "Failed to open %s: %m", src_vc);
+
+        if (is_vconsole(fd) < 0)
+                return log_error_errno(errno, "Device %s is not a virtual console: %m", src_vc);
+
+        if (is_allocated_byfd(fd) < 0)
+                return log_error_errno(errno, "Virtual console %s is not allocated: %m", src_vc);
+
+        if (is_settable(fd) < 0)
```

Comment:
So better to revert those functions to how they were previously - just false / true as their return values ?